### PR TITLE
Provider Teams and Team Measure Results

### PR DESCRIPTION
**Pull request by smc-ssiddiqui**
New features
- User can create and manage teams of providers 
- User can view team measure result - list of providers in a team with their corresponding measure result for a given measure
- User can export spreadsheet report for the team of providers for a given measure
- Teams belong to users specifically, and teams do not cross over between user accounts

Additional:
- fix handlebars gem with working version

### CODE REVIEW

File: app/models/team.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class Team
+  include Mongoid::Document
+  field :name, type: String
+  field :providers, type: Array, default: []
+  field :user_id, type: BSON::ObjectId
```
Comment (eedrummer) [inline app/models/team.rb]: Why is this not a has_and_belongs_to_many? It seems to replicate a lot of the functionality of that macro,

File: app/models/user.rb [hunk 2]
```diff
@@ -84,6 +85,7 @@ class User
   def set_defaults
     self.staff_role ||= APP_CONFIG["default_user_staff_role"]
     self.approved ||= APP_CONFIG["default_user_approved"]
+    self.teams = []
```
Comment (eedrummer) [inline app/models/user.rb]: Isn't this already the default as declared in line 63?

File: app/controllers/admin_controller.rb [hunk 3]
```diff
@@ -49,6 +49,12 @@ def remove_caches
 
   def remove_providers
     Provider.ne('cda_identifiers.root' => "Organization").delete
+    
+    Team.all.each do |team|
```
Comment (eedrummer) [inline app/controllers/admin_controller.rb]: I think this could be Team.all.set(providers: [])

File: app/controllers/practices_controller.rb [hunk 4]
```diff
@@ -73,7 +73,16 @@ def remove_providers
   def destroy
     @practice = Practice.find(params[:id])
     Record.where(practice_id: @practice.id).delete
-    Provider.where(parent_id: @practice.provider.id).delete if @practice.provider
+    if @practice.provider
```
Comment (eedrummer) [inline app/controllers/practices_controller.rb]: Can some of this be replaced with dependent behavior (http://docs.mongodb.org/ecosystem/tutorial/ruby-mongoid-tutorial/#dependent-behaviour)? Maybe using nullify?

**Comment by eedrummer** [comment]
I've made some comments inline. Also, before this is accepted, it will need to pass CI on Ruby > 1.9.3.

File: app/controllers/practices_controller.rb (hunk 4 above)
Comment (smc-ssiddiqui) [inline app/controllers/practices_controller.rb]: yes, you would still need to go through user.teams to delete those providers

**Comment by eedrummer** [comment]
Looks good.

**Comment by petercyli** [comment]
Good to merge.

**Pull request merged**